src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -211,6 +215,28 @@ public void transform(KeyValue<LongWritable, GenericData.Record> input, Emitter<
 
   @Override
   public void onRunFinish(boolean succeeded, BatchSourceContext context) {
+    deleteGcsTemporaryDirectory();
+    deleteBigQueryTemporaryTable();
+  }
+
+  private void deleteBigQueryTemporaryTable() {
+    String temporaryTable = configuration.get(BigQueryConstants.CONFIG_TEMPORARY_TABLE_NAME);
+    if (temporaryTable != null) {
+      String serviceAccount = config.getServiceAccount();
+      Credentials credentials = null;
+      try {
+        credentials = serviceAccount == null ?
```

Comment:
I see this logic at multiple places in this class. Lets create a method and reuse this logic

Reply:
Refactored. A new method got created:

```
private Credentials getCredentials(final String serviceAccount) throws IOException {
    return serviceAccount == null ?
            null : GCPUtils.loadServiceAccountCredentials(serviceAccount, config.isServiceAccountFilePath());
  }
```